src/utils.rs

```diff
@@ -1,33 +1,63 @@
 use crate::AppSW;
 use core::str::from_utf8;
 
-pub const MAX_ALLOWED_PATH_LEN: usize = 10;
+/// BIP32 path stored as an array of [`u32`].
+///
+/// # Generic arguments
+///
+/// * `S` - Maximum possible path length, i.e. the capacity of the internal buffer.
+pub struct Bip32Path<const S: usize = 10> {
+    buffer: [u32; S],
+    len: usize,
+}
 
-/// Convert serialized derivation path to u32 array elements
-pub fn read_bip32_path(data: &[u8], path: &mut [u32]) -> Result<usize, AppSW> {
-    // Check input length and path buffer capacity
-    if data.is_empty() || path.len() < data.len() / 4 {
```

Comment:
A minor bug was here, as the path length encoded in the first byte of `data` is still there.
Correct expression would be `data.is_empty() || (path.len() - 1) < data.len() / 4`.